Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp

```diff
@@ -117,10 +117,20 @@ void DNSResolveQueueCFNet::performDNSLookup(const String& hostname, Ref<Completi
     nw_parameters_set_context(parameters.get(), context.get());
     RetainPtr resolver = adoptCF(nw_resolver_create_with_endpoint(hostEndpoint.get(), parameters.get()));
 
-    nw_resolver_set_update_handler(resolver.get(), dispatch_get_main_queue(), makeBlockPtr([resolver, completionHandler = WTFMove(completionHandler)] (nw_resolver_status_t status, nw_array_t resolvedEndpoints) mutable {
+    auto timer = makeUnique<Timer>([resolver, completionHandler]() {
+        WTF::ensureOnMainThread([resolver, completionHandler]() {
+            nw_resolver_cancel(resolver.get());
+            completionHandler->complete(makeUnexpected(DNSError::Cancelled));
+        });
+    });
+    timer->startOneShot(Seconds(60));
+
+    nw_resolver_set_update_handler(resolver.get(), dispatch_get_main_queue(), makeBlockPtr([resolver, completionHandler = WTFMove(completionHandler), timer = WTFMove(timer)] (nw_resolver_status_t status, nw_array_t resolvedEndpoints) mutable {
         if (status == nw_resolver_status_in_progress)
             return;
-        auto callCompletionHandler = [resolver = WTFMove(resolver), completionHandler = WTFMove(completionHandler)](DNSAddressesOrError&& result) mutable {
+
+        auto callCompletionHandler = [resolver, completionHandler = WTFMove(completionHandler), timer = WTFMove(timer)](DNSAddressesOrError&& result) mutable {
+            timer->stop();
```

Comment:
May be nice to rename this timer to `timeoutTimer` to make it clearer what it's for.